resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DbReader.scala

```diff
@@ -94,6 +95,40 @@ object DbReader {
           """
       .query[KubernetesCluster]
 
+  // We are calculating the grace period for nodepool deletion assuming that the following are valid proxies for an app's last activity:
+  //    1. destroyedDate for deleted apps
+  //    2. createdDate for error'ed apps
+  // Note that we explicitly check nodepools with 5/11 of statuses that exist.
+  // The statuses we exclude are PROVISIONING, STOPPING, DELETED, PRECREATING, PREDELETING, PREDELETING
+  //    - We exclude all -ING statuses because they are transitionary, and the purpose of this is not to handle timeouts
+  //    - Unclaimed we exclude because we never want to clean up batch created nodepools
+  // We are excluding default nodepools, as these should remain for the lifetime of the cluster
+  // TODO: Read the grace period (hardcoded to '1 HOUR' below) from config
+  val applessNodepools =
```

Comment:
@kyuksel @rtitle any comments on the query condition?

Reply:
LGTM 👍🏼